common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/MicrosoftAuthorizationRequest.java

```diff
@@ -22,65 +22,46 @@
 // THE SOFTWARE.
 package com.microsoft.identity.common.internal.providers.microsoft;
 
-import android.net.UrlQuerySanitizer;
+import android.net.Uri;
+import android.os.Build;
 import android.support.annotation.NonNull;
 import android.util.Base64;
 
-import com.microsoft.identity.common.adal.internal.util.StringExtensions;
-import com.microsoft.identity.common.exception.ClientException;
-import com.microsoft.identity.common.exception.ErrorStrings;
+import com.google.gson.annotations.SerializedName;
 import com.microsoft.identity.common.internal.logging.Logger;
-import com.microsoft.identity.common.internal.providers.microsoft.microsoftsts.MicrosoftStsAuthorizationRequest;
+import com.microsoft.identity.common.internal.net.ObjectMapper;
 import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationRequest;
 import com.microsoft.identity.common.internal.providers.oauth2.PkceChallenge;
 import com.microsoft.identity.common.internal.util.StringUtil;
 
 import java.io.UnsupportedEncodingException;
 import java.net.URL;
 import java.nio.charset.Charset;
-import java.util.HashMap;
-import java.util.List;
 import java.util.Map;
-import java.util.Set;
-import java.util.StringTokenizer;
 import java.util.UUID;
 
-import static com.microsoft.identity.common.adal.internal.util.StringExtensions.isNullOrBlank;
-import static com.microsoft.identity.common.adal.internal.util.StringExtensions.urlFormDecode;
-
-public abstract class MicrosoftAuthorizationRequest extends AuthorizationRequest {
+public abstract class MicrosoftAuthorizationRequest<T extends MicrosoftAuthorizationRequest<T>> extends AuthorizationRequest<T> {
     /**
      * Serial version id.
      */
     private static final long serialVersionUID = 6873634931996113294L;
 
-    /* Constants */
     private static final String TAG = MicrosoftAuthorizationRequest.class.getSimpleName();
-    public static final String ENCODING_UTF8 = "UTF_8";
-    public static final String CODE_CHALLENGE = "code_challenge";
-    public static final String CODE_CHALLENGE_METHOD = "code_challenge_method";
-    public static final String QUERY_PROMPT = "prompt";
-    public static final String QUERY_PROMPT_VALUE = "login";
-    public static final String LOGIN_HINT = "login_hint";
-
-    public static final String LIB_ID_PLATFORM = "x-client-SKU";
-    public static final String LIB_ID_VERSION = "x-client-Ver";
-    public static final String LIB_ID_CPU = "x-client-CPU";
-    public static final String LIB_ID_OS_VER = "x-client-OS";
-    public static final String LIB_ID_DM = "x-client-DM";
 
     /**
      * Required.
      */
-    private URL mAuthority;
+    private URL mAuthority; //Not going to be serialized into request url parameters
```

Comment:
we should like apply the transient modifier here to ensure that it's not serialized.